lib/Service/PdfParserService.php

```diff
@@ -39,7 +39,7 @@ private function getDataDir(): string {
 	 */
 	public function getMetadata(string $filePath): array {
 		$fullPath = $this->getDataDir() . $filePath;
-		$json = shell_exec($this->cliPath . ' info ' . $fullPath);
+		$json = shell_exec($this->cliPath . ' info "' . escapeshellcmd($fullPath) . '"');
```

Comment:
I believe it is better to use `escapeshellarg`, i.e.
```
shell_exec(escapeshellarg($this->cliPath) . ' info ' . escapeshellarg($fullPath));
```
because `escapeshellcmd` only escapes `"`s if they are not paired. One can still execute arbitrary commands by using a filename like `my"custom $(rm -rf /)" file.pdf`.

Reply:
The property `cliPath` come from `libresign_cli_path` app value and is defined by application not by user, but, I will check the impact.